astropy/utils/compat/optional_deps.py

```diff
@@ -30,7 +30,9 @@
     "lxml",
     "matplotlib",
     "mpmath",
+    "narwhals",
     "pandas",
+    "polars",
     "PIL",
```

Comment:
(I'm assuming this should be alpha-sorted in a case insensitive way)
```suggestion
    "PIL",
    "polars",
```